requirements/common_constraints.txt

```diff
@@ -25,8 +25,3 @@ django-simple-history==3.0.0
 # tox>4.0.0 isn't yet compatible with many tox plugins, causing CI failures in almost all repos.
 # Details can be found in this discussion: https://github.com/tox-dev/tox/discussions/1810
 tox<4.0.0
-
-# edx-sphinx-theme is not compatible with latest Sphinx==6.0.0 version 
-# Pinning Sphinx version unless the compatibility issue gets resolved
-# For details, see issue https://github.com/openedx/edx-sphinx-theme/issues/197
-sphinx<6.0.0
```

Comment:
This stood out to me when I was looking through this PR so I'm leaving a comment to provide context

* This is the only mention of `sphinx` in this repo https://github.com/search?q=repo%3Aopenedx%2FRecommenderXBlock%20sphinx&type=code
* `edx-sphinx-theme` is deprecated https://github.com/openedx/public-engineering/issues/200
* latest `sphinx` on pypi as of writing is `7.2.6` https://pypi.org/project/Sphinx/